tests/test_az.py

```diff
@@ -0,0 +1,74 @@
+"""Tests for the Azure MultiPartUpload class."""
+
+import unittest
+from unittest.mock import MagicMock, patch
+
+from odc.geo.cog._az import MultiPartUpload
+
```

Comment:
skip this test if `azure` is not available.

Reply:
Also added flags the HAVE_AZURE flag here.